### DISCUSSION TITLE
Package versions

### DISCUSSION THREAD

**Discussion by MatthijsTUD** [capytaine/capytaine#237] [Q&A] [answered]
Hello there,

I'm currently trying to get my tutorial script running (the one on the Capytaine website). It ran fine 2 weeks ago but today I keep getting an error (mkl-service package failed to import). I've tried to rebuild the project since I suspect some of the packages having the wrong version. How do I know which versions are compatible with Capytaine? 

Thank you,

Matthijs

**Comment by mancellin** [comment]
Capytaine should be compatible with all supported version of Python and Numpy (currently that is Python >3.8 and Numpy >1.20, I believe).

MKL-service seems to be a dependancy of Numpy. I am not aware of any issue with it. Could you share some more details for me to investigate?

**Comment by MatthijsTUD** [comment]
Dear Matthieu

Thank you for the prompt reply. Here is the full error message:

C:\Users\matth\anaconda3\envs\pythonProject2\python.exe C:\Users\matth\PycharmProjects\pythonProject2\Capytaine_Tutorial_Sphere.py 
Traceback (most recent call last):
  File "C:\Users\matth\PycharmProjects\pythonProject2\Capytaine_Tutorial_Sphere.py", line 3, in <module>
    from capytaine import *
  File "C:\Users\matth\anaconda3\envs\pythonProject2\lib\site-packages\capytaine\__init__.py", line 21, in <module>
    from capytaine.bem.solver import Nemoh, BEMSolver
  File "C:\Users\matth\anaconda3\envs\pythonProject2\lib\site-packages\capytaine\bem\solver.py", line 24, in <module>
    from capytaine.green_functions.delhommeau import Delhommeau
  File "C:\Users\matth\anaconda3\envs\pythonProject2\lib\site-packages\capytaine\green_functions\delhommeau.py", line 15, in <module>
    import capytaine.green_functions.Delhommeau_f90 as Delhommeau_f90
ImportError: DLL load failed while importing Delhommeau_f90: The specified module could not be found.

Process finished with exit code 1

I'm currently using Python 3.9 and Numpy 1.23.4
Thank you in advance,

Matthijs

**Reply by mancellin** [reply]
`Delhommeau_f90` is the compiled Fortran core of Capytaine. Not being able to load this compiled DLL is unfortunately a common issue on Windows (e.g. #111, #131).
I don't know the source of the problem, but it can usually be solved by creating a new `conda` environment for Capytaine.

**Reply by MatthijsTUD** [reply]
Thank you, I have already made a new environment and this is the error I'm getting (this was the original error):

C:\Users\matth\anaconda3\envs\Thesis_Capytaine\python.exe "C:\Users\matth\PycharmProjects\Thesis_Capytaine\Tutorial Capytaine.py" 
C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\numpy\__init__.py:164: UserWarning: mkl-service package failed to import, therefore Intel(R) MKL initialization ensuring its correct out-of-the box operation under condition when Gnu OpenMP had already been loaded by Python process is not assured. Please install mkl-service package, see http://github.com/IntelPython/mkl-service
  from . import _distributor_init
Traceback (most recent call last):
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\numpy\core\__init__.py", line 22, in <module>
    from . import multiarray
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\numpy\core\multiarray.py", line 12, in <module>
    from . import overrides
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\numpy\core\overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\matth\PycharmProjects\Thesis_Capytaine\Tutorial Capytaine.py", line 3, in <module>
    from capytaine import *
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\capytaine\__init__.py", line 10, in <module>
    from capytaine.meshes.geometry import Axis, Plane, xOz_Plane, yOz_Plane, xOy_Plane
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\capytaine\meshes\__init__.py", line 6, in <module>
    from capytaine.meshes.meshes import Mesh
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\capytaine\meshes\meshes.py", line 12, in <module>
    import numpy as np
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\numpy\__init__.py", line 166, in <module>
    from . import core
  File "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\lib\site-packages\numpy\core\__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.9 from "C:\Users\matth\anaconda3\envs\Thesis_Capytaine\python.exe"
  * The NumPy version is: "1.21.4"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'

Thank you in advance

**Reply by mancellin** [reply]
It seems to be an issue with the installation of Numpy, unrelated to Capytaine. Does it also happen when you create a new conda environment with Numpy but not Capytaine?

It could be an issue with your Pycharm setup as suggested in the URL.

Not sure if relevant, but by default `conda` should install the latest version of a library in a new environment, but here only the version 1.21.4 of Numpy has been installed (the latest version today is 1.23.3; both should work equally fine with Capytaine).

**Comment by MatthijsTUD** [comment]
I'm currently reinstalling PyCharm and creating a new environment with the correct packages through the Conda terminal (as suggested on the Capytaine website). I will let you know if the same error occurs.

**Comment by MatthijsTUD** [comment, marked as answer]
I've removed PyCharm and all installed packages for my projects and did a fresh install of Capytaine and some of the optional packages (vtk and matlibplot). I've ran the tutorial script again and now it works perfectly. I suspect some of my packages which I installed through an unconventional method messed up the compatibility.
